docs/third-party-developers/extensibility/rest-api/extend-rest-api-add-custom-fields.md

```diff
@@ -0,0 +1,243 @@
+- The [parent attribute](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#parent) specifies the parent block that this block should be nested within it. It determines where the block will render. In our example, the block is a child of the `woocommerce/checkout-shipping-methods-block`. This means that your block will be rendered within the `woocommerce/checkout-shipping-methods-block`. If the shipping methods block is not required, your block will not be rendered.
+
+### 3. Setting custom checkout data
+
+We can set the added field data to send it to the `wc/store/checkout` endpoint when processing orders using the function `setExtensionData`:
+
+```JavaScript
+setExtensionData(
+	'namespace-of-your-block',
+	'key-of-your-data',
+	value
+);
+```
+
+#### Parameters
+
+- namespace `string` - The namespace of your block.
+- key `string` - The key of your data.
+- value `any` - The value of your data.
+
+#### How it works
+
+ 1. `setExtensionData` is passed to inner blocks via props.
+ 2. It updates the `extensionData` key of the `wc/store/checkout` data store.
+ 3. This key is passed as part of the request body when POSTing to the checkout endpoint.
+
+#### Code Example
+
+```JavaScript
+// block.js
+export const Block = ( { checkoutExtensionData, extensions } ) => {
+/**
+	 * setExtensionData will update the wc/store/checkout data store with the values supplied. It
+	 * can be used to pass data from the client to the server when submitting the checkout form.
+	 */
+	const { setExtensionData } = checkoutExtensionData;
+}
+
+// ... Some code here
+
+useEffect( () => {
+/**
+	* This code should use `setExtensionData` to update the `key-of-your-data` key
+	* in the `namespace-of-your-block` namespace of the checkout data store.
+*/
+setExtensionData(
+	'namespace-of-your-block',
+	'key-of-your-data',
+	value
+);
+}, [ setExtensionData, value ] );
+```
+
+#### Screenshots
+
+Screenshots of Redux Dev tool showing the data store before and after the setExtensionData call:
+
+| Before | After |
+| ------ | ----- |
+|    <img width="713" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/14235870/948581f5-fdc2-4df1-963f-9aeb4b18b042">    |    <img width="723" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/14235870/ddc7dbe7-3fad-44cd-bd19-ce78bc49b951">   |
+
+### 4. Processing the Checkout POST Request
+
+To process the added field data, we'll need extend the Store API to tell it to expect additional data. A good place to start this process off is within the `initialize` function of our `IntegrationInterface` class.
```

Comment:
I checked on this recently, and actually the `IntegrationInterface` shouldn't be loaded on requests where blocks might not render (e.g. store API requests).

I'd therefore recommend creating a new class specifically for extending the Store API and not doing this as part of `IntegrationInterface`.

Reply:
@opr, do you mean that we shouldn't use the following code since it implements `IntegrationInterface`?

- The `custom-inner-block-blocks-integration.php` file:

```php
use Automattic\WooCommerce\Blocks\Integrations\IntegrationInterface;

/**
 * Class for integrating with WooCommerce Blocks
 */
class Custom_Inner_Block_Blocks_Integration implements IntegrationInterface {
	// ... Some code here

	/**
	 * When called invokes any initialization/setup for the integration.
	 */
	public function initialize() {
		require_once __DIR__ . '/custom-inner-block-extend-store-endpoint.php';
		// ... Some code here: (e.g. init functions that registers scripts and styles, and other instructions)
		$this->extend_store_api();
	}

	/**
	 * Extends the cart schema to include the custom-inner-block value.
	 */
	private function extend_store_api() {
		Custom_Inner_Block_Extend_Store_Endpoint::init();
	}
}
```

We execute `$this->extend_store_api();` inside the `initialize` function. Is this the part that we should move to a new Class?

Reply:
> We execute `$this->extend_store_api();` inside the `initialize` function. Is this the part that we should move to a new Class?

exactly!

Reply:
@opr, I didn't create a new Class for extending the Store API. I think we can use the existing `Custom_Inner_Block_Extend_Store_Endpoint` and initialize it on the plugin php file. I updated the description of our PHP files and added links to our documentation for more details.